system/jlib/jstring.cpp

```diff
@@ -2482,3 +2501,49 @@ const char * nullText(const char * text)
     if (text) return text;
     return "(null)";
 }
+
+StringBuffer& StringBuffer::operator=(StringBuffer&& value)
+{
+    swapWith(value);
+    return *this;
+}
+
+StringBuffer operator+ (const StringBuffer& lhs, const StringBuffer& rhs)
```

Comment:
I'm far from convinced. We have steered clear of operator overloading in our usage because it typically obscures what is actually going on and has led to unwitting creation of temporaries that impact efficiency or (worse) lingering references to temporaries. The new c++11 features may make it easier to write operator overloads that avoid those issues, but they still leave the code harder for an old fogey like me to actually know what is happening. I like to understand what temporaries are in use where, and given that desire the use of operator overloading actually makes the code harder to follow rather than easier.

That might change if we started using them more, I suppose...